payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncher.kt

```diff
@@ -1,95 +1,188 @@
+            readyCallback.onReady(
+                repository.isReady().first().also {
+                    isReady = it
                 }
             )
         }
     }
 
-    fun present() {
-        googlePayController.present()
-    }
+    fun present(clientSecret: String) {
+        check(isReady) {
+            "present() may only be called when Google Pay is available on this device."
+        }
 
-    fun interface ConfigCallback {
-        fun onConfigured(
-            success: Boolean,
-            error: Throwable?
+        activityResultLauncher.launch(
+            GooglePayLauncherContract.Args(
+                clientSecret = clientSecret,
+                config = config
+            )
         )
     }
 
+    @Parcelize
+    internal data class Config @JvmOverloads constructor(
+        val environment: GooglePayEnvironment,
+        val merchantCountryCode: String,
+        val merchantName: String,
+
+        /**
+         * Flag to indicate whether Google Pay collect the customer's email address. Default to `false`.
+         */
+        var isEmailRequired: Boolean = false,
+
+        /**
+         * Billing address collection configuration.
+         */
+        var billingAddressConfig: BillingAddressConfig = BillingAddressConfig(),
+
+        /**
+         * If `true`, Google Pay is considered ready if the customer's Google Pay wallet
+         * has existing payment methods.
+         */
+        var existingPaymentMethodRequired: Boolean = false
+    ) : Parcelable
+
+    @Parcelize
+    internal data class BillingAddressConfig @JvmOverloads constructor(
+        internal val isRequired: Boolean = false,
+
+        /**
+         * Billing address format required to complete the transaction.
+         */
+        internal val format: Format = Format.Min,
+
+        /**
+         * Set to true if a phone number is required to process the transaction.
+         */
+        internal val isPhoneNumberRequired: Boolean = false
+    ) : Parcelable {
+        /**
+         * Billing address format required to complete the transaction.
+         */
+        enum class Format(internal val code: String) {
+            /**
+             * Name, country code, and postal code (default).
+             */
+            Min("MIN"),
+
+            /**
+             * Name, street address, locality, region, country code, and postal code.
+             */
+            Full("FULL")
+        }
+    }
+
+    internal sealed class Result : Parcelable {
+        @Parcelize
+        object Completed : Result()
+
+        @Parcelize
+        data class Failed(
+            val error: Throwable
+        ) : Result()
+
+        @Parcelize
+        object Canceled : Result()
+    }
+
+    internal fun interface ReadyCallback {
+        fun onReady(isReady: Boolean)
+    }
+
     internal fun interface ResultCallback {
-        fun onResult(result: GooglePayLauncherResult)
+        fun onResult(result: Result)
```

Comment:
It seems these interfaces could be collapsed into one

Reply:
I'll keep them separate to follow the API proposal doc.